### PR TITLE
feat:AM-329 Grant explicit access to resource for all(Wildcard)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-329

### Change description ###
feat:AM-329 Grant explicit access to resource for all(Wildcard)

### CODE REVIEW

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 1]
```diff
@@ -863,6 +864,65 @@ public void testReturnResourceAccessListWithUserHavingMultipleRelationshipsOnRoo
...
+                    JsonPointer.valueOf(rootLevelAttribute), ImmutableSet.of(READ)))
+                .accessType(EXPLICIT)
+                .build())
+            .relationships(Collections.emptySet())
+            .build());
+    }
+
+
+    @Test
+    public void filterResourceWithWildcardAccessor() {
```
Comment (shashank-rastogi) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: the method name could be changed to make it more meaningful

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java [hunk 2]
```diff
@@ -47,6 +48,23 @@ public static ExplicitAccessGrant createGrant(String resourceId,
             .build();
     }
 
+    public static ExplicitAccessGrant createGrant(String resourceId,
+                                                  String accessorId,
```
Comment (shashank-rastogi) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: again the name could be more meaningful

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 1 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: done

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java (hunk 2 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: done

**Pull request merged**